generic/juickplugin/juickparser.cpp

```diff
@@ -74,114 +75,96 @@ JuickParser::JuickParser(QDomElement *elem) : elem_(elem)
+    } else if (d->rpostRx.match(msg).hasMatch()) {
         type_     = JM_Reply_Posted;
         infoText_ = QObject::tr("Reply posted.");
-        JuickMessage m(QString(), d->rpostRx.cap(1), QStringList(), QString(), d->rpostRx.cap(2), QString());
+        match     = d->rpostRx.match(msg);
+        JuickMessage m(QString(), match.captured(1), QStringList(), QString(), match.captured(2), QString());
         messages_.append(m);
-    } else if (d->msgPostRx.indexIn(msg) != -1) {
+    } else if (d->msgPostRx.match(msg).hasMatch()) {
         type_     = JM_Message_Posted;
         infoText_ = QObject::tr("New message posted.");
-        JuickMessage m(QString(), d->msgPostRx.cap(1), QStringList(), QString(), d->msgPostRx.cap(2), QString());
+        match     = d->msgPostRx.match(msg);
+        JuickMessage m(QString(), match.captured(1), QStringList(), QString(), match.captured(2), QString());
         messages_.append(m);
-    } else if (d->threadRx.indexIn(msg) != -1) {
+    } else if (d->threadRx.match(msg).hasMatch()) {
         type_     = JM_All_Messages;
         infoText_ = QString();
-        JuickMessage m(d->threadRx.cap(1), d->threadRx.cap(4),
-#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
-                       d->threadRx.cap(2).trimmed().split(" ", Qt::SkipEmptyParts), d->threadRx.cap(3),
-#else
-                       d->threadRx.cap(2).trimmed().split(" ", QString::SkipEmptyParts), d->threadRx.cap(3),
-#endif
-                       d->threadRx.cap(5),
-                       msg.right(msg.size() - d->threadRx.matchedLength() + d->threadRx.cap(6).length()));
+        match     = d->threadRx.match(msg);
+        JuickMessage m(match.captured(1), match.captured(4), match.captured(2).trimmed().split(" ", Qt::SkipEmptyParts),
+                       match.captured(3), match.captured(5),
+                       msg.right(msg.size() - match.capturedLength(0) + match.captured(6).length()));
         messages_.append(m);
-    } else if (d->singleMsgRx.indexIn(msg) != -1) {
+    } else if (d->singleMsgRx.match(msg).hasMatch()) {
         type_     = JM_Post_View;
         infoText_ = QString();
-        JuickMessage m(d->singleMsgRx.cap(1), d->singleMsgRx.cap(4),
-#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
-                       d->singleMsgRx.cap(2).trimmed().split(" ", Qt::SkipEmptyParts), d->singleMsgRx.cap(3),
-#else
-                       d->singleMsgRx.cap(2).trimmed().split(" ", QString::SkipEmptyParts), d->singleMsgRx.cap(3),
-#endif
-                       d->singleMsgRx.cap(6), d->singleMsgRx.cap(5));
+        match     = d->singleMsgRx.match(msg);
+        JuickMessage m(match.captured(1), match.captured(4), match.captured(2).trimmed().split(" ", Qt::SkipEmptyParts),
+                       match.captured(3), match.captured(6), match.captured(5));
         messages_.append(m);
-    } else if (d->juboRx.indexIn(msg) != -1) {
+    } else if (d->juboRx.match(msg).hasMatch()) {
```

Comment:
better try something like 
```
if (match = d->juboRx.match(msg), match.hasMatch()) {
```
or 
```
if ((match = d->juboRx.match(msg)).hasMatch()) {
```
so no need to match again

Reply:
resolved